### PR TITLE
Split the docs into multiple subsystems

### PR DESCRIPTION
**Pull request by addshore**
* Creates a blank template for copying in the future
* alters template readme contents to have links
* adds repo and client components
* split our current docs into repo and client where appropriate
* slightly poke some stakeholders and expectations

### CODE REVIEW

**Review by tzhelyazkova — CHANGES_REQUESTED** [review]
Looks great so far :sparkles:
Adding blank template for copy-pasting is a great idea!
A few suggestions:
- I think it will be better to name the components WikibaseClient and WikibaseRepo given that one of the first tripping points is the difference between them and what is "client" in our context. Also I imagine something like WikibaseLexeme will be added under that name and not just "lexeme". I think those component names will be technical and not conceptiual.
- Let's include WikibaseClient and WikibaseRepo in the general Technical context section already and link them to the folders.

**Comment by addshore** [comment]
> I think it will be better to name the components WikibaseClient and WikibaseRepo given that one of the first tripping points is the difference between them and what is "client" in our context. Also I imagine something like WikibaseLexeme will be added under that name and not just "lexeme". I think those component names will be technical and not conceptiual.

Can do!

> Let's include WikibaseClient and WikibaseRepo in the general Technical context section already and link them to the folders.

Can do!

File: 01-Introduction and Goals.md [hunk 1]
```diff
@@ -4,70 +4,24 @@
...
 
-### Data re-user
+### Small Wikipedia editor
 
-- Access Wikidata's content via a web API.
-- Fix and add values on Wikidata.
+- Have placeholder pages be automatically generated for topics Wikidata contains data for, in order to gain more readers and thus more editors.
 
-#### Researcher
+### Wikidata Researcher
```
Comment (jakobw) [inline 01-Introduction and Goals.md:17]: I liked just "Researcher" better than "Wikidata Researcher". The latter sounds like a person doing research about Wikidata to me?

File: 01-Introduction and Goals.md [hunk 2]
```diff
@@ -4,70 +4,24 @@
...
 - Quickly create visualizations based on data from Wikidata without the need to download it.
-- Fix minor modeling and data issues.
-
-### Data donor
-
-- Import (donate) subset of my collection data to Wikidata.
-- Monitor, maintain, and re-export the data donated to Wikidata.
-- Cross-check an external data collection against Wikidata.
 
 ### Non Wikimedia Wikibase Project
```
Comment (jakobw) [inline 01-Introduction and Goals.md:21]: Do you think this whole section could move into the "repo" component docs? Are any 3rd party Wikibase users interested in client functionality?

File: components/WikibaseRepo/01-Introduction and Goals.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+### Data re-user
+
+- Access a Repos's content via a web API.
+
+#### Wikidata Data re-user
+
+- Fix and add values on Wikidata.
+
+#### Wikidata Researcher
```
Comment (jakobw) [inline components/WikibaseRepo/01-Introduction and Goals.md:51]: Same here, just "Researcher"?

File: 01-Introduction and Goals.md [hunk 4]
```diff
@@ -4,70 +4,24 @@
 
 ## Quality Goals
 
-* Entity data which is included in a client wiki's article content is eventually consistent with updates on the corresponding repo. Data updates from the repo are available to the client within [10 minutes](https://gerrit.wikimedia.org/r/plugins/gitiles/operations/puppet/+/e3423c8627067a8de8352b9b7f3c59870d9b47c5/modules/icinga/manifests/monitor/wikidata.pp#10).
```
Comment (jakobw) [inline 01-Introduction and Goals.md:7]: I like that this turned into one quality goal each for client and repo. Much clearer that way!

**Review by tzhelyazkova — APPROVED** [review]
:sparkles: :sparkles: :sparkles:

**Review by jakobw — COMMENTED** [review]
Generally makes sense to me!! :+1:

File: 01-Introduction and Goals.md (hunk 1 above)
Comment (addshore) [reply 01-Introduction and Goals.md:17]: Will change this back!

File: 01-Introduction and Goals.md (hunk 2 above)
Comment (addshore) [reply 01-Introduction and Goals.md:21]: Yes there are many people that use the client functionality (using wikibase data in wikitext pages).
So it probably needs to stay here for now?

File: components/WikibaseRepo/01-Introduction and Goals.md (hunk 3 above)
Comment (addshore) [reply components/WikibaseRepo/01-Introduction and Goals.md:51]: Will do!

**Review by Silvan-WMDE — CHANGES_REQUESTED** [review]
I generally agree and think that it will make things a lot easier to describe and to understand if we split the docs and fill in multiple arc42 templates for different parts of the system.

In his "High level context" diagram, Adam has already identified five such systems that are probably worth creating separate arc42 templates for:
- Structured Data Repo/Wikidata
- Client
- Query Service
- MediaRepo/Commons
- Tools, such as Quickstatements

However, I would suggest to not call those different parts "components", because that would be very misleading. Section 3 of the arc42 template ("System scope and context") refers to it as "system", having communication partners, neighboring systems and users. It should be interesting to try and create "Business Context and scope" diagrams for each of the above, to make sure we have identified them correctly.

Maybe we can rename the PR into "Create separate docs for separate systems" or even find a better term than "system"?

**Comment by manicki** [comment]
systems sounds good enough to me. Of course could be argued it is also ambiguous, but I think it is less loaded than "component".
Another alternative, also not great and ambiguous could be maybe "context" - although this could make some unwanted connection to "bounded contexts" which likely these are not. I am not sure how much of a problem we see it as of now - i.e. how DDD-savvy we see our engineers to be, as the audience of this documentation.

**Review by Silvan-WMDE — APPROVED** [review]

**Pull request merged**